getFNA.ps1

```diff
@@ -0,0 +1,234 @@
+    
+}
+
+function updateFNA ()
+{
+    checkGit
+    Write-Output "Updating to the latest git version of FNA..."
+
+    git -C "${PSScriptRoot}\FNA" pull --recurse-submodules
+    
+    if ($? -eq 1) { Write-Output "Finished updating!" }
+    else { Write-Output "ERROR: Unable to update." exit}
+}
+
+function check7zip ()
+{
+    if ((Test-Path "C:\Program Files\7-Zip") -eq 0)
+    {
+        Write-Output "ERROR: 7zip is not installed, please install 7zip and try again."
+        exit
+    }
+}
+
+function getLibs ()
+{
+    Write-Output "Downloading the latest FNAlibs..."
+    Invoke-WebRequest -Uri http://fna.flibitijibibo.com/archive/fnalibs.tar.bz2 -OutFile "${PSScriptRoot}/fnalibs.tar.bz2"
+    if ($? -eq 1) { Write-Output "Finished downloading!" }
+    else { Write-Output "ERROR: Unable to download successfully." exit}
+
+    Write-Output "Decompressing fnalibs..."
+    check7zip
+    if ((Test-Path "${PSScriptRoot}\fnalibs") -eq 0)
+    {
+        & "C:\Program Files\7-Zip\7z.exe" x "fnalibs.tar.bz2"
+        if ($? -eq 1){ Remove-Item "fnalibs.tar.bz2"} 
+        else { Write-Output "ERROR: Unable to decompress successfully." exit }
+        & "C:\Program Files\7-Zip\7z.exe" x "fnalibs.tar" -ofnalibs
+        if ($? -eq 1)
+        {  
+            Remove-Item "fnalibs.tar" 
+            Write-Output "Finished decompressing!"
+        } 
+        else { Write-Output "ERROR: Unable to decompress successfully." exit }
+    }
+    
+}
+
+function checkMsbuild ()
+{
+    try { msbuild | Out-Null }
+    catch [System.Management.Automation.CommandNotFoundException] 
+    {
+        Write-Output "ERROR: 'msbuild' is not available. Attempting to look for build tools..."
+        if ((Test-Path "C:\Program Files (x86)\Microsoft Visual Studio") -eq 1) 
+        { 
+            $files = Get-ChildItem -Path "C:\Program Files (x86)\Microsoft Visual Studio\*" -Recurse -Include "msbuild.exe" | Where-Object { (Split-Path (Split-Path $_.FullName -Parent) -Leaf) -like 'bin'}
+            if ($files.Length -gt 0)
+            {
+                if ($files.Length -gt 1)
+                {
+                    Write-Output "Found multiple build tools... unsure which version to use."
+                    for ($i = 0; $i -lt $files.Length; $i++)
+                    {
+                        $filename = $files[$i].FullName
+                        $c = $i + 1
+                        Write-Output "${c}: ${filename}"
+                    }
+                    $max = $files.Length
+                    $choice = Read-Host -Prompt "Choose desired msbuild version? (1-${max})"
+                    if ($choice -le 0 -or $choice -gt $files.Length+1) 
+                    { 
+                        "ERROR: invalid choice, exiting"
+                    }
+
+                    $msbuildpath = Split-Path $files[$choice-1].FullName
+                    Write-Output "your choice was ${choice} which equals this path: ${msbuildpath}. attempting to add to PATH..."
```

Comment:
Is modifying the users PATH really what we want here? Would it not be better to just proceed with the msbuild version indicated for the duration of this script?

Reply:
I am not 100% sure, but I believe that will only change the path for the currently running terminal session, so it shouldn't have any far-reaching affects.

Reply:
After testing, it errors out in an unelevated prompt, and in an elevated one it incorrectly appends to the system PATH (it doesn't add the semicolon seperator)

Reply:
Apologies if this isn't the direction it should go. I had it adding the semicolon seperator and it was causing issues with my machine (not correctly appending to the list in the explorer)

I thought about just adding it to the path for the session, but then it would error every time the build tasks were executed later on in VSCode, which to me is kind of pointless, as if using full visual studio the msbuild issue wouldn't be a problem really. 

I think the only other option would be to ask the user to install the tools and then prompt for a path to the tools and store that path for use later to run msbuild.exe

Reply:
> Are you on Windows 10?
Yup, I am.

Personally I think it's ok to prompt the user to ensure MSBuild is on the PATH. I'm not a fan of scripts like this mucking about with my system settings.

Reply:
Alright I'll remove the msbuild stuff, no problem!